extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -63,28 +62,348 @@ import {
+            dispatch(makeAccountActive(account.publicKey));
+          }
+
+          // save the details of the `accountToSign`
+          autoSelectedAccountDetails = account;
+        }
+      }
+
+      // create an object so we don't need to keep iterating over allAccounts when we switch accounts
+      allAccountsMap.current[account.publicKey] = account;
+    });
+
+    if (!autoSelectedAccountDetails) {
+      setAccountNotFound(true);
+    }
+  }, [accountToSign, allAccounts, dispatch]);
+
+  if ("blob" in blobOrTx) {
+    return (
+      <SignBlobBody setIsDropdownOpen={setIsDropdownOpen} setIsPasswordRequired={setIsPasswordRequired} accountNotFound={accountNotFound} isConfirming={isConfirming} isDropdownOpen={isDropdownOpen} allAccounts={allAccounts} currentAccount={currentAccount} publicKey={publicKey} isPasswordRequired={isPasswordRequired} rejectAndClose={rejectAndClose} handleApprove={handleApprove} setStartedHwSign={setStartedHwSign} isHardwareWallet={isHardwareWallet} hwStatus={hwStatus} isExperimentalModeEnabled={isExperimentalModeEnabled} blob={blobOrTx} t={t} dispatch={dispatch} />
+    )
+  }
+
+  return (
+    <SignTxBody
+      setCurrentAccount={setCurrentAccount}
+      setAccountNotFound={setAccountNotFound}
+      startedHwSign={startedHwSign}
+      setIsDropdownOpen={setIsDropdownOpen} setIsPasswordRequired={setIsPasswordRequired} accountNotFound={accountNotFound} isConfirming={isConfirming} isDropdownOpen={isDropdownOpen} allAccounts={allAccounts} currentAccount={currentAccount} publicKey={publicKey} isPasswordRequired={isPasswordRequired} rejectAndClose={rejectAndClose} handleApprove={handleApprove} setStartedHwSign={setStartedHwSign} isHardwareWallet={isHardwareWallet} hwStatus={hwStatus}
+      isExperimentalModeEnabled={isExperimentalModeEnabled}
+      tx={blobOrTx}
+      t={t}
+      dispatch={dispatch}
+      networkName={networkName}
+      networkPassphrase={networkPassphrase}
+     />
+  )
+};
+
+interface SignBlobBodyProps {
+  publicKey: string
+  currentAccount: Account
+  allAccounts: Account[]
+  isPasswordRequired: boolean
+  isDropdownOpen: boolean
+  isHardwareWallet: boolean
+  isConfirming: boolean
+  accountNotFound: boolean
+  hwStatus: ShowOverlayStatus
+  isExperimentalModeEnabled: boolean
+  blob: BlobToSign
+  t: any
+  dispatch: AppDispatch
+  setStartedHwSign: (hasStarted: boolean) => void
+  setIsPasswordRequired: (isRequired: boolean) => void
+  setIsDropdownOpen: (isRequired: boolean) => void
+  rejectAndClose: () => void
+  handleApprove: (signAndClose: () => Promise<void>) => () => Promise<void>
+}
+
+const SignBlobBody = ({ publicKey, allAccounts, isDropdownOpen, handleApprove, isConfirming, accountNotFound, rejectAndClose, currentAccount, setIsDropdownOpen, setIsPasswordRequired, isPasswordRequired, blob, isExperimentalModeEnabled, t, dispatch, hwStatus, isHardwareWallet, setStartedHwSign }: SignBlobBodyProps) => {
+  const { accountToSign, domain, isDomainListedAllowed } = blob
```

Comment:
looks like you need to run prettier on this file to clean up some of the formatting

Reply:
oh whoops, I expected it to run on commit 🤔 I'll run it manually